main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -385,12 +385,26 @@ object Safety {
             visit(exp)
         }
 
-      case e@Expr.UncheckedCast(exp, _, _, _, _, _) =>
-        val errors = verifyUncheckedCast(e)
-        visit(exp) ++ errors
+      case e@Expr.UncheckedCast(exp, _, _, _, _, loc) =>
+        val ctx = loc.security
+        if (ctx == SecurityContext.AllPermissions) {
+          // Permitted
+          val errors = verifyUncheckedCast(e)
+          visit(exp) ++ errors
+        } else {
+          // Forbidden
+          SafetyError.Forbidden(ctx, loc) :: Nil
+        }
 
-      case Expr.UncheckedMaskingCast(exp, _, _, _) =>
-        visit(exp)
+      case Expr.UncheckedMaskingCast(exp, _, _, loc) =>
+        val ctx = loc.security
+        if (ctx == SecurityContext.AllPermissions) {
+          // Allowed...
```

Comment:
Why not use the same wording as elsewhere? i.e. Permitted / Forbidden?

Reply:
No good reason in all honesty. Mostly me being silly after finding out what `UncheckedmaskingCast`was.
I almost feel like this should never really be allowed, as in it should not be a thing, given it seems like an "unchecked unchecked cast", but there may be some uses I haven't been able to spot?